tests/loader_unknown_ext_tests.cpp

```diff
@@ -43,9 +44,70 @@ class UnknownFunction : public ::testing::Test {
  is called, will return the custom_physical_device_function if the function name matches vkNotRealFuncTEST. The test then calls the
  function to verify that the unknown physical device function dispatching is working correctly.
 */
+template <typename DispatchableHandleType>
+struct custom_functions {
+    static VKAPI_ATTR uint32_t VKAPI_CALL func_zero(DispatchableHandleType handle, uint32_t foo) { return foo; };
+    static VKAPI_ATTR uint32_t VKAPI_CALL func_one(DispatchableHandleType handle, uint32_t foo, uint32_t bar) { return foo + bar; };
+    static VKAPI_ATTR float VKAPI_CALL func_two(DispatchableHandleType handle, uint32_t foo, uint32_t bar, float baz) {
+        return baz + foo + bar;
+    };
+    static VKAPI_ATTR int VKAPI_CALL func_three(DispatchableHandleType handle, int* ptr_a, int* ptr_b) { return *ptr_a + *ptr_b; };
+    static VKAPI_ATTR float VKAPI_CALL func_four(DispatchableHandleType handle, int* ptr_a, int* ptr_b, int foo, int bar, float k,
+                                                 float l, char a, char b, char c) {
+        return *ptr_a + *ptr_b + foo + bar + k + l + static_cast<int>(a) + static_cast<int>(b) + static_cast<int>(c);
+    };
+};
+
+template <typename FunctionStruct>
+void fill_custom_functions(std::vector<VulkanFunction>& driver_function_list, std::vector<std::string>& fake_function_names,
+                           FunctionStruct const& funcs, uint32_t function_count) {
+    for (uint32_t i = 0; i < function_count;) {
```

Comment:
Both of these functions don't check internal to the loop if you've exceeded your function count, potential memory overwrite scenario.  At the very least, verify that the count is a modulus of 5.

Reply:
This is so that the code is flexible for the tests. Tests should manually verify that they wont go out of bounds.